### PR TITLE
Refactor String light identifiers to actual HueLamps

### PR DESCRIPTION
**Pull request by RickWieman**
With #3 in mind:
- Add methods for maintaining the current set of lamps
- Add methods for getting lamp(s) by identifier
- Move "all" identifier handling to the web controller (_should we maybe deprecate this?_)
- Update test suite accordingly

Not yet tested on actual lamps using the Philips Hue facade.

### CODE REVIEW

**Comment by sanderploegsma** [comment]
> Move "all" identifier handling to the web controller (should we maybe deprecate this?)

Already gave my view on this: yes please. When you want to have all lights do something, you should just pass a list of all the lights to a function, the string `all` makes no sense tbh. If the service can expose the available lights to the controller, this shouldn't be too hard.

**Comment by RickWieman** [comment]
It isn't indeed, but I didn't want to break the API in a refactor branch...

File: src/main/java/ch/wisv/chue/HueService.java [hunk 1]
```diff
@@ -30,52 +32,41 @@
...
     /**
-     * Load a state on the lights
+     * Load a state on the lamps
      *
      * @param state            the state to load
-     * @param lightIdentifiers the lights to apply the state to
+     * @param lamps the lamps to apply the state to
      */
-    public void loadState(HueState state, String... lightIdentifiers) {
+    public void loadState(HueState state, List<HueLamp> lamps) {
```
Comment (praseodym) [inline src/main/java/ch/wisv/chue/HueService.java]: `Set<HueLamp>` would be nicer

File: src/main/java/ch/wisv/chue/HueService.java [hunk 2]
```diff
@@ -87,36 +78,62 @@ public void loadEvent(HueEvent event, int duration, String... lightIdentifiers)
...
+        return hueFacade.getAvailableLamps().get(id);
+    }
+
+    /**
+     * @return list with all Hue lamps
      */
-    public List<HueLamp> getAvailableLamps() {
-        return hueFacade.getAvailableLamps();
+    public List<HueLamp> getLamps() {
+        return hueFacade.getAvailableLamps().entrySet().stream()
```
Comment (praseodym) [inline src/main/java/ch/wisv/chue/HueService.java]: [`return hueFacade.getAvailableLamps().values();`](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#values--)?

Comment (RickWieman) [inline src/main/java/ch/wisv/chue/HueService.java]: Oooops :see_no_evil:

File: src/main/java/ch/wisv/chue/HueService.java (hunk 1 above)
Comment (RickWieman) [inline src/main/java/ch/wisv/chue/HueService.java]: Ah, you're totally right, I'll change that.

File: src/main/java/ch/wisv/chue/HueService.java [hunk 3]
```diff
@@ -87,36 +78,62 @@ public void loadEvent(HueEvent event, int duration, String... lightIdentifiers)
...
+    /**
+     * Fetch multiple lamps using a list of identifiers
+     *
+     * @param ids the identifiers of the requested lamps
+     * @return the lamps
+     */
+    public List<HueLamp> getLampsById(List<String> ids) {
+        List<HueLamp> lamps = new ArrayList<>();
+
+        ids.forEach(id -> lamps.add(hueFacade.getAvailableLamps().get(id)));
```
Comment (praseodym) [inline src/main/java/ch/wisv/chue/HueService.java]: I guess you could stream, map, collect here?

**Comment by praseodym** [comment]
Besides that, :shipit: :+1:

**Comment by praseodym** [comment]
lgtm, but let's test before merging :)

**Pull request merged**